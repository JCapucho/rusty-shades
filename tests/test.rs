use rusty_shades::build_naga_ir;
use std::fs::read_to_string;

#[test]
fn insta_test() {
    insta::glob!("*.rsh", |file| {
        let code = read_to_string(file).unwrap();
        let res = build_naga_ir(&code);
        insta::assert_ron_snapshot!(res);
    });
}
