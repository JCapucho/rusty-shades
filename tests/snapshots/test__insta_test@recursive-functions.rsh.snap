---
source: tests/test.rs
expression: res
input_file: tests/recursive-functions.rsh
---
Err([
  Error(
    msg: "Recursive function detected",
    spans: [
      Range(Loc(48), Loc(54)),
      Range(Loc(33), Loc(39)),
      Range(Loc(57), Loc(60)),
    ],
    hints: [],
  ),
])
